stdlib/public/Concurrency/Task.cpp

```diff
@@ -62,10 +171,17 @@ static void completeTask(AsyncTask *task, ExecutorRef executor,
   // to wait for the object to be destroyed.
   _swift_task_alloc_destroy(task);
 
+  if (task->isFuture()) {
+    // Complete the future, taking the result from the context.
+    task->completeFuture(context);
+
+    // Schedule tasks that are waiting on the future.
+    task->scheduleWaitingTasks(executor);
```

Comment:
As mentioned, these need to be combined so that you can deal with the queue atomically.

Reply:
Done, thanks!